### PR TITLE
[VLM] Force FP32 only for Qwen3-Omni CodePredictor

### PR DESCRIPTION
**Pull request by byungilm**
Description
### Issue
The Qwen3-Omni speech pipeline currently applies the FP32 inference precision hint to all six speech submodels when running on GPU.

The precision requirement was isolated to the CodePredictor. Running CodePredictor with the default GPU FP16 precision can change its discrete code predictions and result in degenerate or incorrect speech generation.

However, forcing FP32 on the other speech models is unnecessary. In particular, forcing Code2Wav to FP32 produces an attenuated and incorrect waveform.

### Root Cause
All speech submodels used the same model-loading function, which unconditionally added:

[INFERENCE_PRECISION_HINT=f32](vscode-file://vscode-app/c:/Program%20Files/Microsoft%20VS%20Code/93cfdd489c/resources/app/out/vs/code/electron-browser/workbench/workbench.html)

for every GPU compilation.

This prevented the Talker, embedding, projection, and Code2Wav models from using the caller or GPU plugin default precision.

### Solution
This change makes the FP32 override optional per speech sub-model:
* Force INFERENCE_PRECISION_HINT=f32 only for CodePredictor on GPU.
* Keep the caller/plugin defaults for:
** Thinker text embeddings
** Talker
** Talker text embeddings
** Talker projections
** Code2Wav
* Preserve all caller-provided compilation properties.
* Leave CPU behavior unchanged.
* Avoid introducing environment-variable-based precision overrides.

### Why CodePredictor sub-model must be fp32
After the Layer-2 MLP, code_predictor model produces an outlier of approximately 49K and a normalization dynamic range of up to 597.2×. At this magnitude, FP16 has a ULP of 32, allowing up to approximately 16 units of rounding error and causing smaller residual updates to disappear. The affected tensor is repeatedly processed by residual addition and RMSNorm, then propagated through recurrent attention. Meanwhile, the final sampling margin can be as small as 0.062.

Because the model is autoregressive, one changed code modifies the next embedding and KV state, turning a local rounding error into persistent sequence divergence. This behavior is observed empirically: full FP16 execution diverges on all 296 evaluated frames, whereas preserving FP32 along the sensitive path reduces the discrepancy to 4 frames.

Therefore, selected nodes must execute and materialize their outputs in FP32. This is not a general preference for higher precision; it is a model-specific numerical requirement determined by the measured activation range, FP16 resolution, normalization behavior, sampling margin, and recurrent error propagation.

### ticket
CVS-190210

## Checklist:
- [x] This PR follows [GenAI Contributing guidelines](https://github.com/openvinotoolkit/openvino.genai?tab=contributing-ov-file#contributing). 
- [x] Tests have been updated or added to cover the new code. 
- [x] This PR fully addresses the ticket. 
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: src/cpp/src/visual_language/qwen3_omni/speech_pipeline.cpp [hunk 1]
```diff
@@ -94,19 +94,23 @@ Qwen3OmniSpeechPipeline::Qwen3OmniSpeechPipeline(const std::filesystem::path& mo
...
 
-        // Force FP32 inference precision on GPU for talker models to match CPU behavior
-        // GPU FP16 causes numerical differences in logits → different sampled tokens → wrong speech length
         ov::AnyMap compilation_props = properties;
-        if (device == "GPU" || device.find("GPU") == 0) {
+        if (force_fp32 && (device == "GPU" || device.find("GPU") == 0)) {
             compilation_props["INFERENCE_PRECISION_HINT"] = "f32";
-            GENAI_DEBUG("Speech: forcing FP32 precision for %s on GPU", filename.c_str());
+            GENAI_DEBUG("Speech: forcing f32 precision for %s on GPU", filename.c_str());
         }
```
[the remark this answers is not in the document]
Comment (byungilm) [reply src/cpp/src/visual_language/qwen3_omni/speech_pipeline.cpp:114]: Fixed in db12538.

**Comment by byungilm** [comment]
@sgonorov Hi, this is solution for the CVS-190210 ticket you have submitted. 
Could you please join review?

**Comment by byungilm** [comment]
Hi @sgonorov, 
This PR is for the ticket CVS-190210. 
Could you please check the comment `Why CodePredictor sub-model must be fp32` and review this PR?

File: tests/cpp/test_qwen3_omni_speech_pipeline.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023-2026 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+// Regression tests for Qwen3-Omni speech pipeline CodePredictor GPU precision fix.
+//
+// Case 1 — GPU device detection: device.find("GPU") != std::string::npos
+// Case 2 — Precision hint type:  compilation_props["INFERENCE_PRECISION_HINT"] = ov::element::f32
+
```
[the remark this answers is not in the document]
Comment (byungilm) [reply tests/cpp/test_qwen3_omni_speech_pipeline.cpp]: Acknowledged. The [load_model](vscode-file://vscode-app/c:/Users/byungilm/AppData/Local/Programs/Microsoft%20VS%20Code/8a7abeba6e/resources/app/out/vs/code/electron-browser/workbench/workbench.html) lambda is constructor-local and not directly testable without mocking ov::Core or real model files, so these tests validate the underlying logic patterns instead. A pipeline-level test that verifies per-submodel properties will be tracked as a follow-up refactor.

File: tests/cpp/test_qwen3_omni_speech_pipeline.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023-2026 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+// Regression tests for Qwen3-Omni speech pipeline CodePredictor GPU precision fix.
```
[the remark this answers is not in the document]
Comment (byungilm) [reply tests/cpp/test_qwen3_omni_speech_pipeline.cpp]: Added [tests/cpp/test_qwen3_omni_speech_pipeline.cpp](vscode-file://vscode-app/c:/Users/byungilm/AppData/Local/Programs/Microsoft%20VS%20Code/8a7abeba6e/resources/app/out/vs/code/electron-browser/workbench/workbench.html) covering GPU device string detection (Case 1) and INFERENCE_PRECISION_HINT type correctness (Case 2). Checklist updated accordingly.

**Comment by byungilm** [comment]
@Wovchena  Could you please review this PR ?(also for Qwen3-omni model)

**Review by Wovchena — APPROVED** [review]

**Review by sgonorov — APPROVED** [review]
